### PR TITLE
fix: WorkflowSteps enables completely consistent actions/checkout and setGitIdentity in Workflows

### PR DESCRIPTION
**Pull request by dkershner6**
Fixes https://github.com/projen/projen/issues/3203

Placed a bit of an opinion on WorkflowSteps with a common interface, please weigh in on how you like the pattern.  I moved an unrelated Step to the new interface in order to demonstrate.

This is non-breaking PR, but does include a deprecation, since I moved something.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/github/task-workflow.ts [hunk 1]
```diff
@@ -199,16 +199,10 @@ export class TaskWorkflow extends GithubWorkflow {
         ...preCheckoutSteps,
 
         // check out sources.
-        {
-          name: "Checkout",
-          uses: "actions/checkout@v3",
-          ...(Object.keys(checkoutWith).length > 0
```
Comment (mrgrain) [inline src/github/task-workflow.ts:205]: Huh, so this didn't work then?

**Review by mrgrain — COMMENTED** [review]
Looks good. I think my initial concern was that a single `JobStep` as return type is quite risky, in case it ever needs to be more than just one step. And that led to the invention of `WorkflowActions` which is a admittedly a bit confusing.

**Comment by dkershner6** [comment]
> Looks good. I think my initial concern was that a single JobStep as return type is quite risky, in case it ever needs to be more than just one step. And that led to the invention of WorkflowActions which is a admittedly a bit confusing.

@mrgrain , I actually started this PR with the same opinion, and I understand the inclination entirely.  I changed my opinion abruptly the moment I tried to implement a common interface for a WorkflowStep's "other" fields (id, etc) on something that allowed for more than one step.

I now believe there should be both, and that WorkflowActions can be more opinionated, and WorkflowSteps should be quite generic, so everyone has the building blocks for consistency no matter what they do Workflow wise.

**Review by mrgrain — APPROVED** [review]

**Comment by mrgrain** [comment]
Makes sense. I think long-term we probably want to get away from the struct interfaces and implement these as classes that can be either a single step or multiple steps.

**Pull request merged**